src/frontend/src/expr/type_inference/func.rs

```diff
@@ -603,8 +603,22 @@ fn infer_type_for_special(
                 _ => Ok(None),
             }
         }
+        ExprType::ArrayDims => {
+            ensure_arity!("array_dims", | inputs | == 1);
+            if inputs[0].is_untyped() {
+                return Ok(None);
```

Comment:
It is hit, but I agree the error message reads a little inconsistent. The `Ok(None)` here uses a trick to defer error reporting until it finds `#[function("array_dims(list) -> varchar")]` from the function signature table, while some other functions (like `array_ndims`, `array_distinct`, `array_length`) return `ErrorCode::BindError` directly. The type checking logic here also worth a refactor (e.g. `ensure_array_type(inputs[0])?;`), but this PR intends to clarify the behaviors of these 5 functions for nested list (rather than tensor) first.

Reply:
#10257